src/pages/Detail/Detail.js

```diff
@@ -0,0 +1,196 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import BottomModal from '../../components/BottomModal/BottomModal';
+import Footer from '../../components/Footer/Footer';
+import './Detail.scss';
+
+class Detail extends React.Component {
+  constructor() {
+    super();
+    this.mainTab = React.createRef();
+    this.state = {
+      productInfo: [],
+      modal: false,
+      isClicked: true,
+      isVisible: true,
+    };
+  }
+
+  componentDidMount = () => {
+    window.addEventListener('scroll', this.handleScroll);
+    fetch(`/data/Project.json`)
+      .then(res => res.json())
+      .then(res => {
+        this.setState({ productInfo: res });
+      });
+  };
+
+  componentWillUnmount = () => {
+    window.removeEventListener('scroll', this.handleScroll);
+  };
+
+  handleScroll = () => {
+    const headerHeight = window.pageYOffset;
+    const mainTabHeight = this.mainTab.current.offsetTop;
+
+    if (headerHeight > mainTabHeight - 60) {
+      this.setState({ isVisible: false });
+    } else if (headerHeight < mainTabHeight - 60) {
+      this.setState({ isVisible: true });
+    }
```

Comment:
- 아래 같이 수정하면 계속해서 setState 하는 것을 막을 수 있지 않을까요?
```suggestion
    if (headerHeight > mainTabHeight - 60 && this.state.isVisible) {
      this.setState({ isVisible: false });
    } else if (headerHeight < mainTabHeight - 60  && !this.state.isVisible) {
      this.setState({ isVisible: true });
    }
```

Reply:
어렵네요..